clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -234,9 +229,10 @@
  * successful writes are marked as aborted, hence keeping the transactional guarantees.
  * </p>
  * <p>
- * This client can communicate with brokers that are version 0.10.0 or newer. Older or newer brokers may not support
- * certain client features.  For instance, the transactional APIs need broker versions 0.11.0 or later. You will receive an
- * <code>UnsupportedVersionException</code> when invoking an API that is not available in the running broker version.
+ * This client can communicate with brokers that are version 2.1 or newer. Older brokers may not support
+ * certain client features. For instance, {@code sendOffsetsToTransaction} with all consumer group metadata needs broker
+ * versions 2.5 or later. You will receive an <code>UnsupportedVersionException</code> when invoking an API that is not
```

Comment:
@jolshan Is this correct?

Reply:
I was a little confused by this at first, so I did some digging.
This message is due to the change in the signature of `sendOffsetsToTransaction` requiring `ConsumerGroupMetadata groupMetadata` rather than ` String consumerGroupId)` (KIP introducing the conecpt [here](https://cwiki.apache.org/confluence/display/KAFKA/KIP-447%3A+Producer+scalability+for+exactly+once+semantics), and old method signature deprecated in [this](https://github.com/apache/kafka/commit/3805f3706f8f3ebba81b80915c9259590525fb05) commit)

But yes, this is correct. 👍

Reply:
We could just remove the "with all consumer group metadata" comment, since this is the only remaining version of the method. (I guess you can't add offsets to transactions if you are running 2.4 or older)

Reply:
Interesting, not sure if this implication was fully understood. I'll check what other updates are required due to this.

Reply:
Yeah, I don't think it was when it was deprecated.